dwarvesf/icy-backend
# chore: setup db devbox

## Set up local PostgreSQL databases using Devbox

### What does this PR do?
- [x] Adds Devbox configuration for setting up local PostgreSQL databases
- [x] Removes Docker Compose configuration for PostgreSQL
- [x] Updates Makefile targets for database initialization and reset
- [x] Adds scripts for cleaning and initializing PostgreSQL databases
- [x] Updates README with instructions for using Devbox

### Technical Details:
Files changed:
- `.gitignore`: Ignores data directory for PostgreSQL databases
- `Makefile`: Updates `init` and adds `reset` target for database management
- `README.md`: Adds instructions for using Devbox for database setup
- `devbox.json`: Adds PostgreSQL package and environment variables
- `devbox.lock`: Locks PostgreSQL package version
- `docker-compose.yaml`: Removes PostgreSQL service configuration (replaced by Devbox)
- `process-compose.yaml` (new file): Defines PostgreSQL processes for Devbox

New dependencies:
- PostgreSQL package (version defined in `devbox.lock`)

### How to test:
Prerequisites:
- Devbox installed and configured

Steps:
1. Clone the repository
2. Run `make init` to initialize the local PostgreSQL databases
3. Run `make reset` to reset the databases (clean and reinitialize)
4. Verify that the databases are running by checking the output of `devbox services ls`

Expected outcome:
- Two PostgreSQL instances should be running: one for development (`icy_backend_local`) and one for testing (`icy_backend_local_test`)
- The databases should be accessible on the configured ports (`25432` and `35432`)

### Best Practices:
- Separates development and testing databases
- Uses Devbox for consistent local environment setup
- Provides scripts for easy database management

### Edge Case Handling:
- Handles cases where the data directory already exists or is missing
- Ensures proper permissions and ownership for the data directory
